src/dmd/aliasthis.d

```diff
@@ -181,3 +181,45 @@ bool checkDeprecatedAliasThis(AliasThis at, const ref Loc loc, Scope* sc)
     }
     return false;
 }
+
+/**************************************
+ * Check and set 'att' if 't' is a recursive 'alias this' type
+ * Params:
+ *   att = type reference used to detect recursion
+ *   t   = 'alias this' type
+ *
+ * Returns:
+ *   Whether the 'alias this' is recursive or not
+ */
+bool isRecursiveAliasThis(ref Type att, Type t)
+{
+    auto tb = t.toBasetype();
```

Comment:
Isn't this extra? The old lines that are replaced by calls to this function do not call `toBaseType`.

Reply:
It's necessary in all cases. In some previous patch on this issue a crash showed up when using an `enum`.